elements/src/components/pos-dialog/pos-dialog.tsx

```diff
@@ -0,0 +1,38 @@
+import { Component, Host, h, Method } from '@stencil/core';
+
+@Component({
+  tag: 'pos-dialog',
+  styleUrl: 'pos-dialog.css',
+  shadow: true,
+})
+export class PosDialog {
+  private dialog: HTMLDialogElement;
+
+  @Method()
+  async showModal() {
+    this.dialog.showModal();
+  }
+
+  @Method()
+  async close() {
+    this.dialog.close();
+  }
+
+  render() {
+    return (
+      <Host>
+        <dialog ref={el => (this.dialog = el as HTMLDialogElement)}>
+          <header>
+            <span id="title">
+              <slot />
```

Comment:
You need to use named slots as explained in https://stenciljs.com/docs/templating-jsx#slots

This will fix the currently broken layout of the dialog.

Reply:
Thanks. I was expecting positional arguments to work 🤣